### PR TITLE
filmicrgb: add rational function to toe/shoulder

### PR DESCRIPTION
**Pull request by aurelienpierre**
Rational polynomial designed by MrTeatime (https://discuss.pixls.us/t/proposal-for-updated-filmic-curve-parameterisation-in-darktable/23774/101)
This is guaranteed to not over/under-shoot but the contrast is a bit muted near black/white.
It is introduced in addition to soft/hard contrast intents as a "safe" method.

### CODE REVIEW

File: src/iop/filmicrgb.c [hunk 1]
```diff
@@ -173,7 +175,7 @@ typedef struct dt_iop_filmicrgb_params_t
   float black_point_target; // $MIN: 0.000 $MAX: 20.000 $DEFAULT: 0.01517634 $DESCRIPTION: "target black luminance"
   float white_point_target; // $MIN: 0 $MAX: 1600 $DEFAULT: 100 $DESCRIPTION: "target white luminance"
   float output_power;       // $MIN: 1 $MAX: 10 $DEFAULT: 4.0 $DESCRIPTION: "hardness"
-  float latitude;           // $MIN: 0.01 $MAX: 100 $DEFAULT: 25.0
+  float latitude;           // $MIN: 0.01 $MAX: 100 $DEFAULT: 33.0
```
Comment (TurboGit) [inline src/iop/filmicrgb.c:178]: you've changed the default latitude unconditionally, is that intentional ? is that needed for the new safe mode ? if not maybe should be part of another PR, at least I'm not sure it is good to change that default as people are now used to the module... or maybe there is a strong point doing that ?

Comment (aurelienpierre) [reply src/iop/filmicrgb.c:178]: As the name suggests, "safe" is safer as a default because users can blindly push contrast all the way, it will never overshoot.

Comment (TurboGit) [reply src/iop/filmicrgb.c:178]: I missed that the default was also moved to safe mode.

**Review by TurboGit — APPROVED** [review]
All good !
